15-may-17/solutions/simon/lib/look_and_say.ex

```diff
@@ -0,0 +1,26 @@
+defmodule LookAndSay do
+  @moduledoc """
+  Documentation for LookAndSay.
+  """
+
+  @spec generate_look_and_say_sequence(str :: String.t) :: String.t
+  def generate_look_and_say_sequence(str) do
+    Regex.scan(~r/(\d)\1+|(\d)/, str)
+    |> Enum.map(&regex_to_say_representation/1)
+    |> Enum.join
+  end
+
+  @spec regex_to_say_representation(list :: list) :: String.t
+  defp regex_to_say_representation(list) do
+    list
+    |> Enum.reject(fn(x) -> x == "" end)
```

Comment:
So write two `get_string/1`:

```elixir
defp get_string("") do: ""
defp get_string([match, char]) do ...
```

But I wonder if regular expressions are really the right tool for this job... why not something more like this?

https://gist.github.com/samphilipd/f36bbe7a18b0dc3255ab525ad486bd41